src/model.ts

```diff
@@ -0,0 +1,468 @@
+import { VDomModel } from '@jupyterlab/apputils';
+import {
+  ColorPalette,
+  ColorInterpolationSpace,
+  ColorRGBA64,
+  contrastRatio,
+  hslToRGB,
+  rgbToHSL,
+  ColorHSL
+} from '@microsoft/fast-colors';
+
+const black = new ColorRGBA64(0, 0, 0, 1);
+const white = new ColorRGBA64(1, 1, 1, 1);
+const threshold = 7;
+
+function isBlack(color: ColorRGBA64) {
+  let test = 'Undefined';
+  if (color.r === 0.0 && color.g === 0.0 && color.b === 0.0) {
+    test = 'True';
+  } else {
+    test = 'False';
+  }
+  return test;
+}
+
+function isWhite(color: ColorRGBA64) {
+  let test = 'Undefined';
+  if (color.r === 1.0 && color.g === 1.0 && color.b === 1.0) {
+    test = 'True';
+  } else {
+    test = 'False';
+  }
+  return test;
+}
+
+function hexToRGBA(h: string) {
+  let r = '0',
+    g = '0',
+    b = '0';
+
+  // 4 digits
+  if (h.length === 4) {
+    r = '0x' + h[1] + h[1];
+    g = '0x' + h[2] + h[2];
+    b = '0x' + h[3] + h[3];
+
+    // 6 digits
+  } else if (h.length === 7) {
+    r = '0x' + h[1] + h[2];
+    g = '0x' + h[3] + h[4];
+    b = '0x' + h[5] + h[6];
+  }
+  const a = '1';
+  const rgba = [Number(r) / 256, Number(g) / 256, Number(b) / 256, Number(a)];
+  return rgba;
+}
+
+function rgba_string_to_ColorRGBA64(colorValueRGBAstr: number[]) {
+  const colorRGBA64 = new ColorRGBA64(
+    colorValueRGBAstr[0],
+    colorValueRGBAstr[1],
+    colorValueRGBAstr[2],
+    colorValueRGBAstr[3]
+  );
+  return colorRGBA64;
+}
+
+function set_css_color_properties(
+  name: string,
+  values: ColorRGBA64[],
+  start = 0
+) {
+  let counter = start;
+  for (const v of values) {
+    document.body.style.setProperty(`${name}${counter++}`, v.toStringHexRGB());
+  }
+}
+
+function set_css_font_properties(name: string, values: string[], start = 0) {
+  let counter = start;
+  for (const v of values) {
+    document.body.style.setProperty(`${name}${counter++}`, v);
+  }
+}
```

Comment:
You could avoid redundancy using something like

```suggestion
function setCssColorProperties(
  name: string,
  values: ColorRGBA64[],
  start = 0
) {
  setCssProperties(name, values.map(v => v.toStringHexRGB()), start);
}

function setCssProperties(name: string, values: string[], start = 0) {
  let counter = start;
  for (const v of values) {
    document.body.style.setProperty(`${name}${counter++}`, v);
  }
}
```